samples/vboxwrapper/vboxwrapper.cpp

```diff
@@ -767,103 +784,49 @@ int main(int argc, char** argv) {
     retval = pVM->run(current_cpu_time > 0);
     if (retval) {
         // All 'failure to start' errors are unrecoverable by default
+	vboxlog_msg("ERROR: VM failed to start");
         bool   unrecoverable_error = true;
         bool   skip_cleanup = false;
         bool   do_dump_hypervisor_logs = false;
         string error_reason;
         const char*  temp_reason = "";
 
         if (VBOXWRAPPER_ERR_RECOVERABLE == retval) {
-            error_reason =
-                "    BOINC will be notified that it needs to clean up the environment.\n"
-                "    This is a temporary problem and so this job will be rescheduled for another time.\n";
+			error_reason = pVM->get_error(ENV_UNCLEAN);
             unrecoverable_error = false;
-            temp_reason = "VM environment needed to be cleaned up.";
+            temp_reason = pVM->get_error(VM_ENV).c_str();
```

Comment:
Here, what you get from `pVM->get_error` is temporary object which is deleted after the statement. The pointer you get from `c_str` must not outlive the temporary object. You made this mistake multiple times. I'm flagging only this one but correct all of them.